proxy/coco/src/peer/run_state.rs

```diff
@@ -430,4 +565,160 @@ mod test {
 
         assert!(cmds.is_empty(), "expected no command");
     }
+    // TODO(finto): Test that we only issue Query command if Online or Syncing
+    // TODO(finto): Test that we only issue Clone command if Online or Syncing
```

Comment:
The cases where once we come online we start draining requests, either read from persisted state or created while we were offline, need to be addressed as well.

Reply:
Not sure I understand

Reply:
Say we restore a waiting_room from disk and it will take a while to get online, will we be able to start querying requests that have been stored in the persisted waiting room?

Reply:
Ah, ok I'm with you. So the test case would look like:
1. Have a waiting room with created/requested requests inside it
2. Initialise the run state to online
3. See the commands for every one of the requests from step 1.

Right?

Reply:
Sounds like a viable approach. What we want to ensure is that the pending requests are not accidentally advanced to timed out because the peer was offline.

Reply:
I'm not exactly sure how I can test this scenario come to think of it :thinking: fwiw I don't think we can run into this scenario because we only look at query requests if we're online.